lib/tickets/ticket_query_object.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class TicketQueryObject
+  def initialize(params)
+    # @current_user = current_user
+    @search = params[:query]
+    @offset = params[:offset] || 0
+    @limit = params[:limit] || 100
+  end
+
+  def search_by_username_or_by_email
```

Comment:
I think searching Message `content` and `details` is essential for this to work well.

Reply:
Ok, I can add that to the search object and I will think about how I can change the UI.